File: project/server/src/main/java/ru/zagrebin/dto/UserDto.java
package ru.zagrebin.dto;

import lombok.AllArgsConstructor;
import lombok.Data;
import lombok.NoArgsConstructor;

@Data
@NoArgsConstructor
@AllArgsConstructor
public class UserDto {
    private Long id;
    private String username;
    private String email;
    private String displayName;
    private String role;
    private String avatarUrl;
    private int followersCount;
    private int followingCount;
}

File: project/mobile/app/src/main/java/ru/zagrebin/culinaryblog/data/remote/dto/UserDto.kt
package ru.zagrebin.culinaryblog.data.remote.dto

import ru.zagrebin.culinaryblog.model.UserProfile

data class UserDto(
    val id: Long?,
    val username: String?,
    val email: String?,
    val displayName: String?,
    val role: String?,
    val avatarUrl: String?
)

data class UpdateProfileRequest(
    val username: String? = null,
    val email: String? = null,
    val displayName: String? = null,
    val avatarUrl: String? = null
)

fun UserDto.toModel(): UserProfile = UserProfile(
    id = id,
    username = username,
    email = email,
    displayName = displayName,
    role = role,
    avatarUrl = avatarUrl
)
